internal/testrunner/runners/system/runner.go

```diff
@@ -1378,14 +1462,6 @@ func (r *runner) deleteOldDocumentsDataStreamAndWait(ctx context.Context, dataSt
 	if err := deleteDataStreamDocs(ctx, r.options.API, dataStream); err != nil {
 		return fmt.Errorf("error deleting old data in data stream: %s: %w", dataStream, err)
 	}
-	startHits, err := r.getDocs(ctx, dataStream)
-	if err != nil {
-		return err
-	}
-	// First call already reports zero documents
-	if startHits.size() == 0 {
-		return nil
-	}
```

Comment:
No need to keep this logic (the same as line 1398), as every test will use a fresh data stream without any documents written on it.

Reply:
And do we need the `deleteOldDocumentsDataStreamAndWait` method at all then?

Reply:
I'm not 100% sure.

This logic was already present before adding the flags to run system tests in different stages:
https://github.com/elastic/elastic-package/blob/ff60d256311c85d9b140abd395ab4cb4f0fa9550/internal/testrunner/runners/system/runner.go#L621-L641

Considering the three Agent Policies are kept, one scenario that I could think of where it would be better to keep it is when the Elastic Agent from the stack is used, and it is tested twice the same integration.
There is a handler to wipe the data from the data stream in the `tearDownTest`, but this tear down process could fail and leave the data stream with old documents in the following run.

One way that maybe it could be removed, it is also changing the behavior of the system testing when the Elastic Agent from the stack is used. It would mean to update the method `shouldCreateNewAgentPolicyForTest`.

One doubt about this, would the developers/users be using the current data streams created (e.g. logs-nginx.access-ep)? @jsoriano ?

Reply:
> One way that maybe it could be removed, it is also changing the behavior of the system testing when the Elastic Agent from the stack is used. It would mean to update the method

Yes, it would be nice if when using the Agent from the stack, we also use a clean data stream while testing. This way we would have the same logic independently of the agent used:
* Create the policy for testing.
* Assign the policy to the agent.
* Run the tests.
* Assign the previously assigned policy to the agent.
* Remove the data stream created by the test policy.

I guess that in this case we need at most two policies:
* When `--setup` is used, the package policy can be assigned to the enroll policy. Then the test policy when using `--no-provision` would be basically a copy of the same policy, but with different namespace.
* When the stack agent is used, it already has a previous policy.

> One doubt about this, would the developers/users be using the current data streams created (e.g. logs-nginx.access-ep)? @jsoriano ?

Not sure of understanding the question. When `--setup` is used, the deployment should be functional, so I guess that in this case yes, users would be using these data streams. In other cases I don't think so, as the policies will be configured for specific packages only during tests.

Reply:
> Yes, it would be nice if when using the Agent from the stack, we also use a clean data stream while testing. This way we would have the same logic independently of the agent used:
> 
> * Create the policy for testing.
> * Assign the policy to the agent.
> * Run the tests.
> * Assign the previously assigned policy to the agent.
> * Remove the data stream created by the test policy.

:+1: 


> * When `--setup` is used, the package policy can be assigned to the enroll policy. Then the test policy when using `--no-provision` would be basically a copy of the same policy, but with different namespace.

The enroll policy does not contain any package policy, is just to be able to enroll the agent. In `--setup` stage, it is needed to create another policy that will be assigned back to the agent every time `--no-provision` finishes.

The two policies created (in --setup and --no-provision) are created from the contents of the package. It could be changed by the developer locally.

> Not sure of understanding the question. When `--setup` is used, the deployment should be functional, so I guess that in this case yes, users would be using these data streams. In other cases I don't think so, as the policies will be configured for specific packages only during tests.

Those data streams continue being functional, but from now on they will have another namespace (random) in the data stream.